native/core/src/execution/metrics/utils.rs

```diff
@@ -55,60 +64,21 @@ pub fn update_comet_metric(
         Some(metrics.aggregate_by_name())
     };
 
-    update_metrics(
-        env,
-        metric_node,
-        &metrics
-            .unwrap_or_default()
-            .iter()
-            .map(|m| m.value())
-            .map(|m| (m.name(), m.as_usize() as i64))
-            .collect::<Vec<_>>(),
-        metrics_jstrings,
-    )?;
+    // add metrics
+    node_metrics
+        .unwrap_or_default()
+        .iter()
+        .map(|m| m.value())
+        .map(|m| (m.name(), m.as_usize() as i64))
+        .for_each(|(name, value)| {
+            native_metric_node.metrics.insert(name.to_string(), value);
+        });
 
-    unsafe {
-        for (i, child_plan) in spark_plan.children().iter().enumerate() {
-            let child_metric_node: JObject = jni_call!(env,
-                comet_metric_node(metric_node).get_child_node(i as i32) -> JObject
-            )?;
-            if child_metric_node.is_null() {
-                continue;
-            }
-            update_comet_metric(env, &child_metric_node, child_plan, metrics_jstrings)?;
-        }
-    }
-    Ok(())
-}
+    // add children
+    spark_plan.children().iter().for_each(|child_plan| {
+        let child_node = to_native_metric_node(child_plan).unwrap();
```

Comment:
Can we avoid the `unwrap` here?

```suggestion
        let child_node = to_native_metric_node(child_plan)?;
```

Reply:
This change seems to be disallowed.

![image](https://github.com/user-attachments/assets/7197eafe-8adc-426d-a977-7d7b3ec55dd5)